src/test/java/org/prebid/server/bidder/medianet/MedianetBidderTest.java

```diff
@@ -38,8 +46,121 @@ public void setup() {
+                        .build(),
+                mapper.writeValueAsString(
+                        givenBidResponse(bidBuilder -> bidBuilder.impid("123"))));
+
+        //when
+        final Result<List<BidderBid>> result = medianetBidder.makeBids(httpCall, null);
+
+        //then
+        assertThat(result.getErrors()).isEmpty();
+        assertThat(result.getValue()).hasSize(1);
+        assertThat(result.getValue())
+                .contains(BidderBid.of(Bid.builder().impid("123").build(), banner, "USD"));
+    }
+
+    @Test
+    public void makeBidsShouldReturnVideoBidIfVideoIsPresentInRequestImp() throws JsonProcessingException {
+        //given
+        final HttpCall<BidRequest> httpCall = sampleHttpCall(BidRequest.builder()
+                        .imp(singletonList(Imp.builder().id("123").video(Video.builder().build()).build()))
+                        .build(),
+                mapper.writeValueAsString(givenBidResponse(bidBuilder -> bidBuilder.impid("123"))));
+
+        //when
+        final Result<List<BidderBid>> result = medianetBidder.makeBids(httpCall, null);
+
+        //then
+        assertThat(result.getErrors()).isEmpty();
+        assertThat(result.getValue()).hasSize(1);
+        assertThat(result.getValue())
+                .contains(BidderBid.of(Bid.builder().impid("123").build(), video, "USD"));
+    }
+
+    @Test
+    public void makeBidsShouldReturnxNativeBidIfxNativeIsPresentInRequestImp() throws JsonProcessingException {
+        //given
+        final HttpCall<BidRequest> httpCall = sampleHttpCall(BidRequest.builder()
+                        .imp(singletonList(Imp.builder().id("123").xNative(Native.builder().build()).build()))
+                        .build(),
+                mapper.writeValueAsString(givenBidResponse(bidBuilder -> bidBuilder.impid("123"))));
+
+        //when
+        final Result<List<BidderBid>> result = medianetBidder.makeBids(httpCall, null);
+
+        //then
+        assertThat(result.getErrors()).isEmpty();
+        assertThat(result.getValue()).hasSize(1);
+        assertThat(result.getValue())
+                .contains(BidderBid.of(Bid.builder().impid("123").build(), xNative, "USD"));
+    }
+
+    @Test
+    public void makeBidsShouldReturnErrorPreBidExceptionIfNotFoundAnyImpressions() throws JsonProcessingException {
+        //given
+        final HttpCall<BidRequest> httpCall = sampleHttpCall(BidRequest.builder()
+                        .imp(singletonList(Imp.builder().id("123").audio(Audio.builder().build()).build()))
+                        .build(),
+                mapper.writeValueAsString(givenBidResponse(bidBuilder -> bidBuilder.impid("123"))));
+
+        //when
+        final Result<List<BidderBid>> result = medianetBidder.makeBids(httpCall, null);
+
+        //then
+        assertThat(result.getErrors()).hasSize(1);
+        assertThat(result.getErrors())
```

Comment:
calls like `hasSize()` can be chained together, please simplify to smth like:

` assertThat(result.getErrors())
         .hasSize(1);
         .allMatch(..)`
         
Also, please, check for similar occurences.